pkg/internal/itest/testpeer/generator.go

```diff
@@ -259,3 +329,89 @@ func StartAndWaitForReady(ctx context.Context, manager datatransfer.Manager) err
 		return err
 	}
 }
+
+func HttpHandler(ctx context.Context, lsys linking.LinkSystem) func(http.ResponseWriter, *http.Request) {
+	return func(res http.ResponseWriter, req *http.Request) {
+		urlPath := strings.Split(req.URL.Path, "/")[1:]
+
+		// validate CID path parameter
+		cidStr := urlPath[1]
+		rootCid, err := cid.Parse(cidStr)
+		if err != nil {
+			http.Error(res, fmt.Sprintf("Failed to parse CID path parameter: %s", cidStr), http.StatusBadRequest)
+			return
+		}
+
+		// Grab unixfs path if it exists
+		unixfsPath := ""
+		if len(urlPath) > 2 {
+			unixfsPath = "/" + strings.Join(urlPath[2:], "/")
+		}
+
+		// Parse car scope and use it to get selector
+		carScope := types.CarScopeAll
+		if req.URL.Query().Has("car-scope") {
```

Comment:
Currently `RetrievalRequest#GetUrlPath` _always_ appends a `car-scope`, I think as long as we're doing that we should mandate that the query string has to have it, then at least we'll know if it's broken, or if we change policy we have to be clear about it because we break tests.